File: 412.Fizz_Buzz/solution.py
def fizzBuzz(n):
    index = 1
    answer = []
    while index <= n:
        if index % 3 == 0 and index % 5 == 0:
            answer.append("Fizz Buzz")
            index += 1
        elif index % 3 == 0:
            answer.append("Fizz")
            index += 1
        elif index % 5 == 0:
            answer.append("Buzz")
            index += 1
        else:
            answer.append(index)
            index += 1
    return answer


# case 1
print(fizzBuzz(3))

# case 2
print(fizzBuzz(5))

# case 3
print(fizzBuzz(15))

File: 412.Fizz_Buzz/solution.js
// Accepted

/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function (n) {
  let i = 1;
  let answer = [];
  while (i <= n) {
    if (Number.isInteger(i / 3) && Number.isInteger(i / 5)) {
      answer.push("FizzBuzz");
    } else if (Number.isInteger(i / 3)) {
      answer.push("Fizz");
    } else if (Number.isInteger(i / 5)) {
      answer.push("Buzz");
    } else {
      answer.push(String(i));
    }
    i++;
  }
  return answer;
};

// Case 1
console.log(fizzBuzz(3)); // ? ['1', '2', 'Fizz']

// Case 2
console.log(fizzBuzz(5)); // ? ['1', '2', 'Fizz', '4', 'Buzz']

// Case 3
console.log(fizzBuzz(15)); // ? ['1', '2', 'Fizz', '4', 'Buzz', 'Fizz', '7', '8', 'Fizz', 'Buzz', '11', 'Fizz', '13', '14', 'FizzBuzz']
